src/pipecat/services/google/google.py

```diff
@@ -1097,3 +1410,459 @@ async def run_image_gen(self, prompt: str) -> AsyncGenerator[Frame, None]:
+        return language_to_google_stt_language(language) or "en-US"
+
+    async def set_languages(self, languages: List[Language]):
+        """Update the service's recognition languages.
+
+        Args:
+            languages: List of languages for recognition. First language is primary.
+        """
+        logger.info(f"Switching STT languages to: {languages}")
+        self._settings["language_codes"] = [
+            self.language_to_service_language(lang) for lang in languages
+        ]
+        # Recreate stream with new languages
+        if self._streaming_task:
+            await self._disconnect()
+            await self._connect()
+
+    async def set_model(self, model: str):
+        """Update the service's recognition model."""
+        await super().set_model(model)
+        self._settings["model"] = model
+        # Recreate stream with new model
+        if self._streaming_task:
+            await self._disconnect()
+            await self._connect()
+
+    async def start(self, frame: StartFrame):
+        await super().start(frame)
+        await self._connect()
+
+    async def stop(self, frame: EndFrame):
+        await super().stop(frame)
+        await self._disconnect()
+
+    async def cancel(self, frame: CancelFrame):
+        await super().cancel(frame)
+        await self._disconnect()
+
+    async def update_options(
+        self,
+        *,
+        languages: Optional[List[Language]] = None,
+        model: Optional[str] = None,
+        enable_automatic_punctuation: Optional[bool] = None,
+        enable_spoken_punctuation: Optional[bool] = None,
+        enable_spoken_emojis: Optional[bool] = None,
+        profanity_filter: Optional[bool] = None,
+        enable_word_time_offsets: Optional[bool] = None,
+        enable_word_confidence: Optional[bool] = None,
+        enable_interim_results: Optional[bool] = None,
+        enable_voice_activity_events: Optional[bool] = None,
+        location: Optional[str] = None,
+    ) -> None:
+        """Update service options dynamically.
+
+        Args:
+            languages: New list of recongition languages.
+            model: New recognition model.
+            enable_automatic_punctuation: Enable/disable automatic punctuation.
+            enable_spoken_punctuation: Enable/disable spoken punctuation.
+            enable_spoken_emojis: Enable/disable spoken emojis.
+            profanity_filter: Enable/disable profanity filter.
+            enable_word_time_offsets: Enable/disable word timing info.
+            enable_word_confidence: Enable/disable word confidence scores.
+            enable_interim_results: Enable/disable interim results.
+            enable_voice_activity_events: Enable/disable voice activity detection.
+            location: New Google Cloud location.
+
+        Note:
+            Changes that affect the streaming configuration will cause
+            the stream to be reconnected.
+        """
+        needs_reconnect = False
```

Comment:
Do we need this ? It feels like if the user has invoked this method and changed any of the options, we would need to check whether a reconnection is necessary at the end anyway.